include/GFp/versioned.h

```diff
@@ -0,0 +1,400 @@
+#include <generated_versioned.h>
+
```

Comment:
I think it's going to be unreasonable to maintain this file. Let's move each `#define` to be right above the declaration (when in a header file) or definition (when there is no declaration in a header file.) That would make it easier for us to keep everything in sync and it will work better with incremental compilation. In particular when we add/rename a `GFp_BN` symbol, we shouldn't need to recompile any C code other than the bignum code.

In the case of the asm/perlasm-defined code, we may need to duplicate the use of `VERSIONED()` between the C and asm code, but I am fine with that. Overall it would still be easy to maintain. Especially over time there will be less and less C code calling assembly code (I hope) so less and less duplication needed.